packages/teleport/src/components/LabelSelector/LabelSelector.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+
+import { Box, Flex, Pill, Popover, Link, Text } from 'design';
+import { Info, Warning } from '../../../../design/src/Icon';
+
+import { useClickOutside } from 'shared/hooks/useClickOutside';
+import { useEscape } from 'shared/hooks/useEscape';
+
+const VALID_LABEL = /^[a-z]+:\s?[0-9a-z-.]+$/;
+
+function LabelSelector({ onChange }: LabelSelectorProps) {
+  const [labels, setLabels] = useState<string[]>([]);
+  const [showAdd, setShowAdd] = useState(false);
+  const [newLabel, setNewLabel] = useState('');
+  const [validLabel, setValidLabel] = useState(false);
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  const infoIconRef = useRef<HTMLDivElement>();
+  const addLabelInputRef = useRef<HTMLInputElement>();
+  const addLabelContainerRef = useRef<HTMLDivElement>();
+
+  useEffect(() => {
+    setValidLabel(VALID_LABEL.test(newLabel));
+  }, [newLabel]);
+
+  useEffect(() => {
+    onChange(labels);
+  }, [labels]);
+
+  useEffect(() => {
+    if (showAdd && addLabelInputRef.current) {
+      addLabelInputRef.current.focus();
+    }
+  }, [showAdd]);
+
+  useClickOutside(addLabelContainerRef, () => {
+    setShowAdd(false);
+  });
+
+  useEscape(() => {
+    setShowAdd(false);
+  });
+
+  const handleAddLabel = () => {
+    setLabels([...labels, newLabel.trim()]);
+    setNewLabel('');
+  };
+
+  const handleLabelDismiss = (label: string) => {
+    const labelList = [...labels];
+    labelList.splice(labelList.indexOf(label), 1);
+    setLabels(labelList);
+  };
+
+  return (
+    <div>
+      <Heading>
+        <Flex justifyContent="space-between">
+          <Flex>
+            <Text>Assign Labels (optional)</Text>
+            <div
+              ref={infoIconRef}
+              style={{ marginLeft: '12px', float: 'left' }}
```

Comment:
float police, could you check to see other float usage in this file, i see others below

Reply:
🚨 😁 
All updated